EscelForce.Business.Login/Services/MapExtraction/UpdateMapService.cs

```diff
@@ -0,0 +1,159 @@
+                
+                _persistenceContainer?.Set(
+                    BusinessConstants.ChildList, childObjectNames);
+
+                return ServiceResponseModelFactory.GetModel(childObjectNames);
+            }
+            catch (Exception ex)
+            {
+                _loggerManager.LogError(ex.GetExceptionLog());
+
+                return ServiceResponseModelFactory
+                    .GetNullModelForReferenceType<IEnumerable<SfObject>>("An error occurred while fetching child object names");
+            }
+        }
+
+        public ServiceResponseModel<IEnumerable<string>> GetMapNames()
+        {
+            try
+            {
+                var persistentMapNames =
+                      _persistenceContainer.Get<IEnumerable<string>>(BusinessConstants.MapList);
+
+                if (persistentMapNames != null)
+                    return ServiceResponseModelFactory.GetModel(persistentMapNames);
+
+
+                var MapNames = _extractMapRepository.GetRecords().Select(c => c.Name);
+
+                _persistenceContainer?.Set(
+                    BusinessConstants.MapList, MapNames);
+
+                return ServiceResponseModelFactory.GetModel(MapNames);
+            }
+            catch (Exception ex)
+            {
+                _loggerManager.LogError(ex.GetExceptionLog());
+
+                return ServiceResponseModelFactory
+                    .GetNullModelForReferenceType<IEnumerable<string>>("An error occurred while fetching Map names");
+            }
+        }
+
+        public ServiceResponseModel<SfObject> GetObjectNameByMapName(string mapName)
+        {
+            try
+            {
+                var updatedObject =
+                      _persistenceContainer.Get<SfObject>(BusinessConstants.UpdatedObject);
+
+                if (updatedObject != null)
+                    return ServiceResponseModelFactory.GetModel(updatedObject);
+
+                var response = _extractMapRepository.GetRecords().FirstOrDefault(s => s.Name.Equals(mapName));
+                updatedObject = new SfObject
+                {
+                    ApiName = response.Query?.Parent?.ApiName,
+                    Name= response.Query?.Parent?.Label
+
+                };
+
+                _persistenceContainer?.Set(
+                    BusinessConstants.UpdatedObject, updatedObject);
+
+                return ServiceResponseModelFactory.GetModel(updatedObject);
+            }
+            catch (Exception ex)
+            {
+                _loggerManager.LogError(ex.GetExceptionLog());
+
+                return ServiceResponseModelFactory
+                    .GetNullModelForReferenceType<SfObject>("An error occurred while fetching parent object");
+            }
+        }
+
+        public ServiceResponseModel<IEnumerable<string>> GetObjectNames()
```

Comment:
Please remove this method for now, if there is no implementation needed at this point of time.